pkg/instancetype/instancetype_test.go

```diff
@@ -736,6 +736,100 @@ var _ = Describe("Instancetype and Preferences", func() {
 			field = k8sfield.NewPath("spec", "template", "spec")
 		})
 
+		Context("Resources.Requests", func() {
+
+			It("should apply to VMI by using both Instancetype.Resources.Requests.CPU and Instancetype.Resources.Requests.Memory", func() {
+				instancetypeSpec = &instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+					Resources: v1.ResourceRequirements{
+						Requests: k8sv1.ResourceList{
+							k8sv1.ResourceCPU:    resource.MustParse("2"),
+							k8sv1.ResourceMemory: resource.MustParse("512Mi"),
+						},
+					},
+				}
+
+				conflicts := instancetypeMethods.ApplyToVmi(field, instancetypeSpec, preferenceSpec, &vmi.Spec)
+				Expect(conflicts).To(BeEmpty())
+
+				Expect(vmi.Spec.Domain.Resources.Requests).To(Equal(instancetypeSpec.Resources.Requests))
+			})
+
+			It("should apply to VMI by using both Instancetype.Resources.Requests.CPU and InstanceType.Spec.Memory", func() {
+				instancetypeSpec = &instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+					Resources: v1.ResourceRequirements{
+						Requests: k8sv1.ResourceList{
+							k8sv1.ResourceCPU: resource.MustParse("2"),
+						},
+					},
+					Memory: instancetypev1alpha2.MemoryInstancetype{
+						Guest: resource.MustParse("512Mi"),
+					},
+				}
+
+				conflicts := instancetypeMethods.ApplyToVmi(field, instancetypeSpec, preferenceSpec, &vmi.Spec)
+				Expect(conflicts).To(BeEmpty())
+
+				Expect(vmi.Spec.Domain.Resources.Requests).To(Equal(instancetypeSpec.Resources.Requests))
+			})
+
+			It("should apply to VMI by using both Instancetype.Resources.Requests.Memory and InstanceType.Spec.CPU", func() {
+				instancetypeSpec = &instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+					Resources: v1.ResourceRequirements{
+						Requests: k8sv1.ResourceList{
+							k8sv1.ResourceMemory: resource.MustParse("512Mi"),
+						},
+					},
+					CPU: instancetypev1alpha2.CPUInstancetype{
+						Guest: uint32(2),
+					},
+				}
+
+				conflicts := instancetypeMethods.ApplyToVmi(field, instancetypeSpec, preferenceSpec, &vmi.Spec)
+				Expect(conflicts).To(BeEmpty())
+
+				Expect(vmi.Spec.Domain.Resources.Requests).To(Equal(instancetypeSpec.Resources.Requests))
+			})
+
+			It("should return a conflict when Instancetype.Spec.CPU and Instancetype.Spec.Resources.Requests.CPU are both defined", func() {
+				instancetypeSpec = &instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+					CPU: instancetypev1alpha2.CPUInstancetype{
+						Guest: uint32(2),
+					},
+					Resources: v1.ResourceRequirements{
+						Requests: k8sv1.ResourceList{
+							k8sv1.ResourceCPU: resource.MustParse("2"),
+						},
+					},
+				}
+
+				conflicts := instancetypeMethods.ApplyToVmi(field, instancetypeSpec, preferenceSpec, &vmi.Spec)
+
+				Expect(conflicts).To(HaveLen(1))
+				Expect(conflicts[0].String()).To(Equal("spec.template.spec.domain.resources.requests.cpu"))
```

Comment:
So this case needs to be handled by a validation webhook for instancetypes IMHO, we shouldn't be catching it late here.